frontend/src/plaquettes/Footprint.js

```diff
@@ -3,123 +3,194 @@ import Plaquette from './Plaquette';
-    this.templateButton = new Button('Step 1: Define a Template', x, y + 50);
-    this.templateButton.on('click', () => {
-      // Create the template
-      this.renderTemplateControlButtons();
-      this.defineTemplateArea();
+    this.container.name = 'footprint';
+    this.footprintQubits = [];
+    this.initializeFootprint();
+  }
+
+  /**
+   * Initialize the footprint
+   * @function initializefootprint
+   * @returns {void}
+   */
+  initializeFootprint = () => {
+    // Create the footprint
+    this.selectQubitsButton.on('click', () => {
+      this.app.view.addEventListener('click', this.selectQubit);
+      this.unselectQubitsButton.visible = true;
+      this.selectQubitsButton.visible = false;
+      this.createPlaquetteButton.visible = false;
+    });
+    this.unselectQubitsButton.on('click', () => {
+      // Unselect the qubits
+      this.selectedQubits.forEach((qubit) => {
+        qubit.deselect();
+        // For each qubit, remove the text
+        qubit.removeChildren();
+      });
       // Clear the selected qubits
       this.selectedQubits = [];
-      // Hide the button
-      this.templateButton.visible = false;
+      this.selectQubitsButton.visible = true;
+      this.unselectQubitsButton.visible = false;
+      this.createPlaquetteButton.visible = false;
+    });
 
-      // Show a notification to now select qubits within the template to make a plaquette
+    this.createPlaquetteButton.on('click', () => {
+      this.createPlaquette();
     });
-    this.container.addChild(this.templateButton);
-    this.container.addChild(this.plaquetteButton);
-    this.container.name = 'template';
-    // this.workspace.addChild(this.container);
-  }
 
-  // Render the template control buttons
-  renderTemplateControlButtons() {
+    // Initialize the buttons
+    this.unselectQubitsButton.visible = false;
+    this.selectQubitsButton.visible = true;
+    this.createPlaquetteButton.visible = false;
+    this.workspace.addChild(this.selectQubitsButton);
+    this.workspace.addChild(this.unselectQubitsButton);
+    this.workspace.addChild(this.createPlaquetteButton);
+  };
+
+  /**
+   * Select a qubit from the footprint
+   * @function selectQubit
+   * @param {Event} e - The click event
+   * @returns {void}
+   */
+  selectQubit = (e) => {
+    // Check if the click was on a qubit
+    const canvasRect = this.app.view.getBoundingClientRect(); // Get canvas position
+
+    // Calculate the relative click position within the canvas
+    const relativeX = e.clientX - canvasRect.left;
+    const relativeY = e.clientY - canvasRect.top;
+    // Get all the qubits
+    const qubits = this.workspace.children.filter(
+      (child) => child.isQubit === true
+    );
+    const qubit = qubits.find(
+      // Find the qubit that was clicked
+      (q) => q.checkHitArea(relativeX, relativeY) === true
+    );
+    if (!qubit && !(qubit?.isQubit === true)) return; // Check that the qubit exists
```

Comment:
`qubit.isQubit` should be deprecated, since a qubit is always a qubit; perhaps you meant to check its visibility?

Reply:
This is a built in for the QubitClass since it's a convenient way to check if child components, such as graphics or containers are indeed representing a qubit. Without the .isQubit state, it would be difficult to filter through a set of child containers and determine if its a qubit or not.

Reply:
In that case, you should use `instanceof Qubit`.

Reply:
sounds good! thanks